src/kernel/execve/binfmt/elf.rs

```diff
@@ -231,6 +236,30 @@ impl ElfHeader {
     }
 }
 
+/// The loader function for regular elf executable file.
+pub(super) fn load_elf(fs: &FileSystem, parameters: &mut ExecveParameters) -> Result<LoadResult> {
+    // parse LoadInfo from the binary file to be executed
+    let mut load_info = LoadInfo::from(fs, &parameters.host_path)
+        .with_context(|| format!("Failed to parse elf file: {:?}", parameters.host_path))?;
+
+    load_info.raw_path = Some(parameters.raw_guest_path.clone());
+    load_info.user_path = Some(parameters.canonical_guest_path.clone());
+    load_info.host_path = Some(parameters.host_path.clone());
+
+    // An interpreter should not depend on another interpreter
```

Comment:
Any reason why you have this comment here? Do you have any references? Or did this happen from a practical example? e.g., running a command

Reply:
> Any reason why you have this comment here? Do you have any references? Or did this happen from a practical example? e.g., running a command

This comment comes from the C version of proot.
```C
		 /* An ELF interpreter is supposed to be
		 * standalone.  */
```
https://github.com/proot-me/proot/blob/20d619d907e7c80dc33b884a93d0645c7ea96cc2/src/execve/enter.c#L664-L665

Although there is no documentation that explicitly mentions that the ELF interpreter should be statically linked (i.e. no PT_INTERP segment). But from [this article](https://lwn.net/Articles/631631/) on how does kernel load ELF files, it appears that when loading the ELF interpreter, the handler does not handle segments of type `PT_INTERP`, only `PT_LOAD`. 

> After the program code has been loaded into memory as described previously, the ELF handler also loads the ELF interpreter program into memory with load_elf_interp(). This process is similar to the process of loading the original program: the code checks the format information in the ELF header, reads in the ELF program header, **maps all of the PT_LOAD segments from the file into the new program's memory**, and leaves room for the interpreter's BSS segment.

The kernel source code also confirms this 
https://elixir.bootlin.com/linux/v3.18/source/fs/binfmt_elf.c#L452

Also, realistically, as the most common ELF interpreter, `/lib64/ld-linux-x86-64.so.2` is indeed statically linked.
![image](https://user-images.githubusercontent.com/14936301/127585372-5dc2623e-6dec-46ba-b435-96a3fa4e8f23.png)